api/src/Repository/UserRepository.php

```diff
@@ -221,4 +221,70 @@ public function findActiveLaysInLastYear()
 
         return $result->fetchAllAssociative();
     }
+
+    public function getAllAdminAccounts()
+    {
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsCreatedButNotActivatedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
+
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn IS NULL
+                AND u.registrationDate < '{$date}' ";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllActivatedAdminAccounts()
+    {
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn IS NOT NULL";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsNotUsedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
+
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn < '{$date}'  ";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsUsedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
+
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn > '{$date}'  ";
```

Comment:
Would suggest using a parameter here rather than dropping the variable into the string. Even though you define the date above it's a good habit to get into and as the code drifts there's less chance of somebody forgetting that was how it was done and opening up a SQL injection vulnerability. https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/dql-doctrine-query-language.html#named-and-positional-parameters